pyeudiw/federation/trust_chain_validator.py

```diff
@@ -36,17 +52,43 @@ def is_valid(self):
         # inspect the entity statement kid header to know which 
         # TA's public key to use for the validation
         
+        es_payload = unpad_jwt_payload(rev_tc[0])
+        es_exp = es_payload["exp"]
+        
         # if valid: exps.append(this-exp)
+        self._validate_exp(es_exp)
         
-        fed_jwks = unpad_jwt_payload(rev_tc[0])['jwks']['keys']
+        #current_kid = es_payload["jwks"]["keys"][0]["kid"]
+        
+        fed_jwks = es_payload["jwks"]["keys"]
+                
         # for st in rev_tc[1:]:
         # validate the entire chain taking in cascade using fed_jwks
         # if valid -> update fed_jwks with $st
+        
+        for st in rev_tc[1:]:
+            st_header = unpad_jwt_header(st)
+            st_payload = unpad_jwt_payload(st)
+            
+            try:
+                self._validate_keys(fed_jwks, st_header)
+            except Exception as e:
+                logger.warning(f"Warning: {e}")
+                return False
+            
+            try:
+                self._validate_exp(st_payload["exp"])
+            except Exception as e:
```

Comment:
perfavore specializza questa eccezione e aggiungi poi quella generica per ogni eventualità

Reply:
Resolved in [6138769](https://github.com/italia/eudi-wallet-it-python/pull/74/commits/61387698c2272ffb4a363a5bcd6c490e71263736)